mash_client/cli/auth/__init__.py

```diff
@@ -115,6 +116,59 @@ def logout(context):
         echo_style(result['msg'], config_data['no_color'])
 
 
+@click.command()
+@click.option(
+    '--username',
+    type=click.STRING,
+    required=True,
+    help='The username for the mash user.'
+)
+@click.pass_context
+def oidc(context, username):
+    """
+    Handle mash OpenID Connect authentication.
+    """
+    config_data = get_config(context.obj)
+
+    with handle_errors(config_data['log_level'], config_data['no_color']):
+        job_data = {'username': username}
+        result = handle_request(
+            config_data,
+            '/auth/oauth2_req',
+            job_data=job_data,
+            action='post'
+        )
+
+        # display authentication message
+        echo_style(
+            '{}: {}'.format(result['msg'], result['auth_url']),
+            config_data['no_color']
+        )
+
+        auth_code = get_oauth2_code(result['redirect_port'])
+
+        job_data = {
+            'username': username,
```

Comment:
Ditto from server comments. I think the username is not needed here. That info is now coming from the identity server which is requested by the mash server.